File: MyClub/MyClub.Model/Responses/DashboardSalesByCategoryResponse.cs
using System;

namespace MyClub.Model.Responses
{
    public class DashboardSalesByCategoryResponse
    {
        public int CategoryId { get; set; }
        public string CategoryName { get; set; }
        public int TotalSold { get; set; }
        public decimal TotalRevenue { get; set; }
        public decimal Percentage { get; set; }
    }
} 

File: MyClub/UI/myclub_desktop/lib/models/dashboard/dashboard_sales_by_category_response.dart
class DashboardSalesByCategoryResponse {
  final int categoryId;
  final String categoryName;
  final int totalSold;
  final double totalRevenue;
  final double percentage;

  DashboardSalesByCategoryResponse({
    required this.categoryId,
    required this.categoryName,
    required this.totalSold,
    required this.totalRevenue,
    required this.percentage,
  });

  factory DashboardSalesByCategoryResponse.fromJson(Map<String, dynamic> json) {
    return DashboardSalesByCategoryResponse(
      categoryId: json['categoryId'],
      categoryName: json['categoryName'],
      totalSold: json['totalSold'],
      totalRevenue: json['totalRevenue']?.toDouble() ?? 0.0,
      percentage: json['percentage']?.toDouble() ?? 0.0,
    );
  }
}
